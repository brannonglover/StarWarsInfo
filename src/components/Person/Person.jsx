import React from 'react';
import { PersonStyle, PersonStyleH3, PersonStyleP } from './Person.styled';

const Person = ({ person }) => {
  return (
    <PersonStyle>
      <PersonStyleH3>{person.name}</PersonStyleH3>
      <PersonStyleP>Gender - {person.gender}</PersonStyleP>
      <PersonStyleP>Birth year - {person.birth_year}</PersonStyleP>
    </PersonStyle>
  )
}

export default Person;