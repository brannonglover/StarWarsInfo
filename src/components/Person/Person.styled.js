import styled from 'styled-components';

export const PersonStyle = styled.div`
  padding: 8px 16px;
  background: #1b1b1b;
  margin: 16px 0;
  border-radius: 20px;
`;

export const PersonStyleP = styled.p`
  margin: 6px 0;
  color: #999;
`;

export const PersonStyleH3 = styled.h3`
  margin: 10px 0;
  color: #ffff57;
`;