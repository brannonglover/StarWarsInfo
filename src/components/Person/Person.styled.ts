import styled from 'styled-components';

export const PersonStyle = styled.div`
  padding: .5em 1em;
  background: #1b1b1b;
  margin: 1em 0;
  border-radius: 1.2em;
`;

export const PersonStyleP = styled.p`
  margin: .5em 0;
  color: #999;
`;

export const PersonStyleH3 = styled.h3`
  margin: .7em 0;
  color: #ffff57;
`;