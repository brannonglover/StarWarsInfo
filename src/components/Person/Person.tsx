import React from "react";
import { PersonStyle, PersonStyleH3, PersonStyleP } from "./Person.styled";
import strings from "../../common/strings";

interface IPersonProps {
  person: any;
}

const Person: React.FC<IPersonProps> = (props: IPersonProps) => {
  const { person } = props;

  return (
    <PersonStyle>
      <PersonStyleH3>{person.name}</PersonStyleH3>
      <PersonStyleP>
        {strings.GENDER} - {person.gender}
      </PersonStyleP>
      <PersonStyleP>
        {strings.BIRTH_YEAR} - {person.birth_year}
      </PersonStyleP>
    </PersonStyle>
  );
};

export default Person;
