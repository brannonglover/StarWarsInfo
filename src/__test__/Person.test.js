import React from 'react';
import { cleanup, render } from '@testing-library/react';
import Person from '../components/Person/Person';

afterEach(cleanup);

test('Display Person component', () => {
  const person = {
    name: "Luke Skywalker",
    gender: "male",
    birth_year: "052689"
  }
  const { asFragment } = render(<Person person={person} />);

  expect(asFragment()).toMatchSnapshot();
});