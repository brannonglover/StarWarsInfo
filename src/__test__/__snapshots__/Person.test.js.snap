// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display Person component 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ jrrTlY"
  >
    <h3
      class="sc-dlfnbm hvVpLe"
    >
      Luke Skywalker
    </h3>
    <p
      class="sc-gsTCUz bNrjYQ"
    >
      Gender - male
    </p>
    <p
      class="sc-gsTCUz bNrjYQ"
    >
      Birth Year - 052689
    </p>
  </div>
</DocumentFragment>
`;
